### PR TITLE
Fix LoRA inference routing in OpenPipe ART cookbook

### LINKED ISSUE #1649: [Bug] rl_with_openpipe_art: inference targets base model instead of LoRA adapter during GRPO training
### Version

1.4.0

### Which installation method(s) does this occur on?

_No response_

### Summary
The `rl_with_openpipe_art` cookbook's `config.yml` sets `openpipe_llm.model_name` to `Qwen/Qwen2.5-3B-Instruct`, but ART registers the LoRA adapter under `backend.name` (`tic_tac_toe_training_run`). Every epoch, NAT runs inference against the base model instead of the latest LoRA checkpoint, making GRPO training ineffective.

### How ART Routing Works
Per the ART client docs, inference must use `model.get_inference_name()` (the training run name), not the base model name, so that:
> "Each new LoRA instance will be automatically loaded onto the vLLM server... inference requests are always routed to the latest version of the model."

### The Bug
```
# config.yml (abbreviated)
llms:  
  openpipe_llm:    
     model_name: Qwen/Qwen2.5-3B-Instruct  # always hits base model

trainer_adapters:  
   openpipe_trainer_adapter:    
      backend:      
         name: "tic_tac_toe_training_run"      # LoRA adapter registered here      
         base_model: "Qwen/Qwen2.5-3B-Instruct"
```

Rewards are always computed from the unchanged base model. The RL loop cannot converge.

### Impact
- Training is a no-op: GRPO updates LoRA weights that are never queried for the next evaluation.
- Misleading to users: Stagnant reward curves appear to be a tuning problem rather than a config bug.

### Why Full-Weight Training Would Mask This Bug
The README states ART uses "TorchTune for GRPO weight updates," but the config specifies no `torchtune_args`, so ART falls back to Unsloth LoRA finetuning. With full-weight training, vLLM updates weights in-place under the same model name, so the mismatch wouldn't matter.

Note: In the current ART API (v0.5.11), `TorchtuneArgs` has been replaced by TinkerArgs. Full-weight training can alternatively be enabled through init_args.full_finetuning:
```
trainer_adapters:  
   openpipe_trainer_adapter:    
      backend:      
         init_args:        
            max_seq_length: 8192        
            full_finetuning: true  # enables full-weight training via Unsloth
```
The NAT plugin is pinned to ART 0.4.11 which still references `art.dev.TorchtuneArgs`. Regardless, the README should clarify that the current config uses LoRA by default, and explain the implications for model naming.

### Suggested Fix
1. `config.yml` and `config_post_train.yml` -- change model_name to match `backend.name`:
```
llms:  
   openpipe_llm:    
      model_name: tic_tac_toe_training_run  # must match backend.name for LoRA routing
```
2. README -- clarify that the default config uses Unsloth LoRA (not TorchTune full-weight), and that `model_name` must match `backend.name` when using LoRA.

### Affected Files
- examples/finetuning/rl_with_openpipe_art/src/rl_with_openpipe_art/configs/config.yml
- examples/finetuning/rl_with_openpipe_art/src/rl_with_openpipe_art/configs/config_post_train.yml
- examples/finetuning/rl_with_openpipe_art/README.md

### Minimum reproducible example

```shell

```

### Relevant log output

<details><summary>Click here to see error details</summary><pre>

 [Paste the error here, it will be hidden by default]

</pre></details>

### Other/Misc.

Avg rewards per epoch before and after fixing the inference pointer;

Before:
Image

After:
Image

Note: the big drop at the final epoch may be related to some vLLM float reporting issues, which lies more on the OpenPipe and/or vLLM side. Needs more investigation.

### Code of Conduct

- [x] I agree to follow the NeMo Agent Toolkit Code of Conduct
- [x] I have searched the [open bugs](https://github.com/NVIDIA/NeMo-Agent-Toolkit/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

### PR DESCRIPTION
**Pull request by aslanshi**
### Summary
- Fix `model_name` in training and post-training configs to point to the LoRA adapter name instead of the base model, so GRPO training actually evaluates the latest checkpoint at each epoch.
- Update README to clarify that when no `torchtune_args` are specified, the default backend uses Unsloth LoRA (not TorchTune full-weight training) and document model name routing requirements.
- Sync README inline config snippet with actual `config.yml` values.

### Problem
During LoRA finetuning, the ART backend registers each LoRA adapter in vLLM under the training run name (`tic_tac_toe_training_run`). However, the `openpipe_llm` config had `model_name: Qwen/Qwen2.5-3B-Instruct`, causing every inference request to hit the unchanged base model. Rewards were always computed against the base model, making GRPO training ineffective.

This bug it's not likely to occur with full-weight training since vLLM updates weights in-place under the same model name.

### Fix
- `config.yml`: Changed `model_name` from `Qwen/Qwen2.5-3B-Instruct` to `tic_tac_toe_training_run`
- `config_post_train.yml`: Same change so post-training evaluation also targets the LoRA adapter
- `README.md`: Corrected "TorchTune" to "Unsloth LoRA" as the default training backend; synced inline config values; added notes explaining LoRA vs. full-weight model name routing

Closes #1649.

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/NeMo-Agent-Toolkit/blob/develop/docs/source/resources/contributing/index.md).
- We require that all contributors "sign-off" on their commits. This certifies that the contribution is your original work, or you have rights to submit it under the same license, or a compatible license.
  - Any contribution which contains commits that are not Signed-Off will not be accepted.
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

## Summary by CodeRabbit

* **Documentation**
  * Clarified that LoRA finetuning with Unsloth is the default and added guidance on model routing and adapter registration for inference.
  * Noted that full-weight training requires extra TorchTune configuration.
  * Updated vocabulary to accept the term "Unsloth".

* **Updates**
  * Adjusted training hyperparameters (learning rate, beta, epochs) and evaluation data reference.
  * Updated training backend reference and config comments to reflect LoRA workflow.

### CODE REVIEW

**Review by dnandakumar-nv — APPROVED** [review]
Good catch, thanks!

**Comment by dnandakumar-nv** [comment]
@aslanshi looks like you're failing a documentation check in CI.

```bash
 examples/finetuning/rl_with_openpipe_art/README.md
 222:92  error  Did you really mean 'Unsloth'?  Vale.Spelling 
 224:46  error  Did you really mean 'Unsloth'?  Vale.Spelling
```

Would you be able to add `Unsloth` to `ci/vale/styles/config/vocabularies/nat/accept.txt` ?

**Comment by aslanshi** [comment]
> @aslanshi looks like you're failing a documentation check in CI.
> 
> ```shell
>  examples/finetuning/rl_with_openpipe_art/README.md
>  222:92  error  Did you really mean 'Unsloth'?  Vale.Spelling 
>  224:46  error  Did you really mean 'Unsloth'?  Vale.Spelling
> ```
> 
> Would you be able to add `Unsloth` to `ci/vale/styles/config/vocabularies/nat/accept.txt` ?

Added.

**Comment by willkill07** [comment]
/ok to test [fe4bdce](https://github.com/NVIDIA/NeMo-Agent-Toolkit/pull/1662/commits/fe4bdce1da821e1b25d74745d68727cd2e2d001d)

**Comment by aslanshi** [comment]
@willkill07 There are some dead links in other places of the project, thus failing the check.

**Comment by aslanshi** [comment]
@willkill07 This is ready to merge now.
